src/libs/ajax/data-table-providers/DataTableProvider.ts

```diff
@@ -64,6 +94,14 @@ export type DeleteTableFn = (entityType: string) => Promise<Response>
 
 export type DownloadTsvFn = (signal: AbortSignal, entityType: string) => Promise<Blob>
 
+export type isInvalidTsvFn = (options: isInvalidTsvOptions) => boolean
+
+export type isTsvUploadButtonDisabledFn = (options: isTsvUploadButtonDisabledOptions) => boolean
+
+export type tsvUploadButtonTooltipFn = (options: tsvUploadButtonTooltipOptions) => string
+
+export type uploadTsvFn = (uploadParams: UploadParameters) => Promise<Response>
```

Comment:
The WDS provider doesn't return a Response any more. And it looks like the Entity Service provider returns a mix of different types.